lib/functoria/impl.ml

```diff
@@ -42,6 +42,17 @@ and 'a device = ('a, abstract) Device.t
 
 let abstract t = Abstract t
 
+let app_has_no_arguments = function
+  | App { args ; _ } ->
+    (match args with
+     | Cons _ -> false
+     | _ -> true)
+  | Dev { args ; _ } ->
+    (match args with
+     | Cons _ -> false
+     | _ -> true)
```

Comment:
Here there is no distinction between `App` and `Dev`. So we can simplify things by deleting lines 46 to 49. But I'm not sure an unikernel using a device without arguments won't also fail (or is supposed to?). I surely have to look deeper into it :)

Reply:
I tried and failed due to GADT and the OCaml compiler complaining about types escaping its scope. But feel free to experiment with that and find a good solution :)

Reply:
see 85886a1a (as proposed by @reynir)